tool/bindings_generator/generate_bindings.dart

```diff
@@ -7,11 +7,48 @@ import 'dart:js_util';
 
 import 'translator.dart';
 import 'util.dart';
+import 'webref_css_api.dart';
 import 'webref_idl_api.dart';
 
+/// Generate CSS property names for setting / getting CSS properties in JS.
+Future<List<String>> _generateCSSStyleDeclarations() async {
+  final cssStyleDeclarations = <String>{};
+  final array = objectEntries(await promiseToFuture<JSObject>(css.listAll()));
+  for (var i = 0; i < array.length.toDart; i++) {
+    final entry = array[i.toJS] as JSArray;
+    final data = entry[1.toJS] as JSObject;
```

Comment:
not necessarily actionable right now: all these index accesses feel very cumbersome, I think we will want to continue iterating on how to improve the experience here.

My default at the moment would be to create a proxy to wrap the array, use regular iteration, and use a Dart int indices.

Reply:
Yea, we definitely want to do something better here.

As you suggest, for places where performance isn't a concern, then an adapter class seems like a good choice.

We can provide such helper adapters, perhaps we can even avoid reifying the adapters on JS backends.